lib/system/io.nim

```diff
@@ -14,6 +14,11 @@ include inclrtl
 import std/private/since
 import formatfloat
 
+when defined(windows) and not defined(useWinAnsi):
+  import os
+  import system/widestrs
+  import terminal
```

Comment:
This is bad, as all modules have `system/io` implicitly imported. This will cause recursive imports, which we don't support (but we should) atm.

Reply:
We can work around **os** by calling **GetLastError**.
We can also work around **terminal** by accessing **File** struct contents (although this may not be pretty).
But working around **widestrs** is another matter...

This is what makes code work across compilers.
GCC has this functionality built-in, but what about other compilers?

Reply:
IIRC console have special HANDLE values that we can just compare to instead.

As for widestrs, you might not even have to import it.

Reply:
https://github.com/nim-lang/Nim/blob/devel/lib/pure/terminal.nim#L669-L672

`isatty()` is simple enough to just duplicate into `io` for this.